docs/source/_notebooks/keras-image-classifiers.rst

```diff
@@ -0,0 +1,435 @@
+
+Explaining Keras image classifier predictions
+=============================================
+
+If we have a model that takes in an image as its input, and outputs
+class scores, i.e. probabilities that a certain object is present in the
+image, then we can use ELI5 to check what is it in the image that made
+the model predict a certain class score. We do that using a method
+called 'Grad-CAM'.
+
+We will be using images from ImageNet (for more information see
+http://image-net.org/), and classifiers from ``keras.applications``.
+
+This has been tested with Python 3.7.3, Keras 2.2.4, and Tensorflow
+1.13.1.
+
+1. Explaining the model's prediction
+------------------------------------
+
+To start out, lets load our image classifier and data.
+
+.. code:: ipython3
+
+    from PIL import Image
+    from IPython.display import display
+    
+    import logging
+    import tensorflow as tf
+    tf.get_logger().setLevel(logging.ERROR) # disable Tensorflow warnings for this tutorial
+    import warnings
+    warnings.simplefilter("ignore") # disable Keras warnings for this tutorial
+    import keras
+    from keras.applications import mobilenet_v2
+    
+    import eli5
+    
+    # let's instantiate our model
+    model = mobilenet_v2.MobileNetV2(include_top=True, weights='imagenet', classes=1000)
+    
+    # and load our sample image
+    image = 'imagenet-samples/cat_dog.jpg'  # TODO: replace this image with something more interesting
+    
+    dims = model.input_shape[1:3] # 224x224
+    doc = eli5.keras.image_from_path(image, image_shape=dims)
+    doc = mobilenet_v2.preprocess_input(doc) # mobilenetv2-specific preprocessing (this operation is in-place)
+    
+    # looking good?
+    display(keras.preprocessing.image.array_to_img(doc[0]))
+
+
+.. parsed-literal::
+
+    Using TensorFlow backend.
+
+
+
+.. image:: ../_notebooks/keras-image-classifiers_files/keras-image-classifiers_1_1.png
+
+
+Let's classify our image and see where the network 'looks' when making
+that classification:
+
+.. code:: ipython3
+
+    eli5.show_prediction(model, doc)
+
+
+.. parsed-literal::
+
+    Taking top prediction: 243
+
+
+
+
+.. image:: ../_notebooks/keras-image-classifiers_files/keras-image-classifiers_3_1.png
+
+
+
+The class ID (index into the output layer) ``243`` stands for
+``bull mastiff`` in ImageNet with 1000 classes
+(https://gist.github.com/yrevar/942d3a0ac09ec9e5eb3a ). Indeed we have a
+dog in the picture. According to Grad-CAM, the area over the dog helped
+the model decide this. Makes sense!
+
+2. Choosing the target class (target prediction)
+------------------------------------------------
+
+We can make the model classify other objects and check where the
+classifier looks to find those objects.
+
+.. code:: ipython3
+
+    cat_idx = 282 # ImageNet ID for "tiger_cat" class, because we have a cat in the picture
```

Comment:
Maybe it would be helpful to show indices and probabilities of top 5 or 10 predictions from the network, I imagine this is the way you obtained these indices, it's not some random cat class but the highest scoring one?

Reply:
It would be cool to include this with the `Explanation` object as well. Perhaps a new attribute like `.output` or `.predictions` would be required?

Reply:
https://github.com/TeamHG-Memex/eli5/pull/315/commits/34e3cee6bb51dabfe867dcaf67888bcd9f0ad30f

Reply:
Now that I think about this again, I wonder if we could use ``TargetExplanation`` for that?

Reply:
`TargetExplanation` has a required `feature_weights` argument. I am not sure if the argument would be applicable in this case. I could make it an optional parameter?